x/hard/keeper/borrow_test.go

```diff
@@ -39,389 +38,580 @@ func (suite *KeeperTestSuite) TestBorrow() {
 		borrowCoins               sdk.Coins
 		expectedAccountBalance    sdk.Coins
 		expectedModAccountBalance sdk.Coins
+		wantDepositErr            string
 	}
 	type errArgs struct {
 		expectPass bool
 		contains   string
 	}
 	type borrowTest struct {
 		name    string
-		args    args
-		errArgs errArgs
+		args    []args
```

Comment:
I think we can refactor this to make it more clear and maintainable with the added multiple borrows

args currently contain a lot of the initial setup so instead of reusing it for following borrows we could split this into something like this:

```go
type setupArgs struct {
	usdxBorrowLimit           sdk.Dec
	priceKAVA                 sdk.Dec
	loanToValueKAVA           sdk.Dec
	priceBTCB                 sdk.Dec
	loanToValueBTCB           sdk.Dec
	priceBNB                  sdk.Dec
	loanToValueBNB            sdk.Dec
	borrower                  sdk.AccAddress
	depositCoins              []sdk.Coin
	// renamed this to be more clear
	initialBorrowCoins        sdk.Coins
	expectedAccountBalance    sdk.Coins
	expectedModAccountBalance sdk.Coins
	wantDepositErr            string
}

type borrowArgs struct {
	borrowCoins sdk.Coins
	// Empty if the borrow should succeed
	wantErr string
}

type borrowTest struct {
	name    string
	setup   setupArgs
	borrows []borrowArgs
}
```

Reply:
I can update as suggested, but the current task is to add new tests to fix a specific bug. However, your suggestions are recommending to going back and fixing how all the tests are running and structured, which is distracting from the main focus.

Reply:
yeah understandable to avoid making changes outside goal of resolving the bug, though I think this still fits with testing this edge case since it requires your addition of multiple borrow cases which changes the structure - just want to make sure any modifications we make are clear and intentional even if it requires some shuffling around in order to support it